### PR TITLE
Adding MSFT recommendation model based on Spark ALS

### PR DESCRIPTION
**Pull request by dciborow**

### CODE REVIEW

**Comment by dciborow** [comment]
Starting Pull request to test build

File: src/text-featurizer/src/test/scala/TextFeaturizerSpec.scala [hunk 1]
```diff
@@ -7,12 +7,12 @@ import com.microsoft.ml.spark.schema.DatasetExtensions._
 import org.apache.spark.ml.feature.{NGram, Tokenizer}
 import org.apache.spark.ml.util.MLReadable
 
-class TextFeaturizerSpec extends TestBase with EstimatorFuzzing[TextFeaturizer]{
```
Comment (mhamilton723) [inline src/text-featurizer/src/test/scala/TextFeaturizerSpec.scala]: Is this just style changes for auto-formatting?

File: src/msft-recommendation/src/main/scala/org/apache/spark/ml/recommendation/msft/MsftRecommendation.scala [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+      instance.userFactors.write.format("parquet").save(userPath)
+      val itemPath = new Path(path, "itemFactors").toString
+      instance.itemFactors.write.format("parquet").save(itemPath)
+    }
+
+  }
+
+  private[MsftRecommendationModel] class MsftRecommendationModelReader extends MLReader[MsftRecommendationModel] {
+
+    /** Checked against metadata when loading model */
```
Comment (mhamilton723) [inline src/msft-recommendation/src/main/scala/org/apache/spark/ml/recommendation/msft/MsftRecommendation.scala]: A lot of this code might be taken care of by ConstructorWritable and ConstructorReadable. These use reflection to generate serialization code so you dont have to mess with writers. The text featurizer has some usage examples.

File: src/text-featurizer/src/test/scala/TextFeaturizerSpec.scala (hunk 1 above)
Comment (dciborow) [reply src/text-featurizer/src/test/scala/TextFeaturizerSpec.scala]: yes... I had auto-format on comment turned on in intellij and must have had the file opened.

File: src/msft-recommendation/build.sbt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+//> DependsOn: core
+version := "1.0"
```
Comment (mhamilton723) [inline src/msft-recommendation/build.sbt]: dont need lines 2 and 3

**Pull request closed without merging**